### PR TITLE
Add `current()` and `currentOrFail()` tenant methods

### PR DESCRIPTION
**Pull request by lukinovec**

### CODE REVIEW

File: src/Database/Models/Tenant.php [hunk 1]
```diff
@@ -45,6 +46,16 @@ public function getTenantKey(): int|string
         return $this->getAttribute($this->getTenantKeyName());
     }
 
+    public static function current(): self|null
```
Comment (stancl) [inline src/Database/Models/Tenant.php]: Unless there's a specific reason for using `self`, always use `static`.

`self` is static with limitations that make sense in certain cases.

File: src/Database/Models/Tenant.php [hunk 2]
```diff
@@ -45,6 +46,16 @@ public function getTenantKey(): int|string
...
+    {
+        return tenant();
+    }
+
+    public static function currentOrFail(): self
```
Comment (stancl) [inline src/Database/Models/Tenant.php]:
```suggestion
    /** @throws TenancyNotInitializedException */
    public static function currentOrFail(): static
```

File: src/Database/Models/Tenant.php (hunk 1 above)
Comment (lukinovec) [reply src/Database/Models/Tenant.php]: Yeah, I used the model name again and cs fixer changed it to `self` after I pushed. I'll change that to `static`

**Comment by lukinovec** [comment]
@stancl I think you can merge this now

**Comment by stancl** [comment]
Thanks!

**Pull request merged**